#! /bin/sh -e
## 011_high-uids.patch by Christian Perrier <bubulle@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Include upstream changes for high UID suppor tin faillog and lastlog
## DP: Debian bug #280212

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@

--- shadow-4.0.3.ori/configure.in	2005-01-10 08:26:18.000000000 +0100
+++ shadow-4.0.3/configure.in	2005-01-10 08:42:08.000000000 +0100
@@ -6,7 +6,7 @@
 dnl Some hacks...
 test "$prefix" = "NONE" && prefix="/usr"
 test "$prefix" = "/usr" && exec_prefix=""
-test "$CFLAGS" = "" && CFLAGS="-O2 -Wall"
+test "$CFLAGS" = "" && CFLAGS="-O2 -Wall -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64"
 test "$LDFLAGS" = "" && LDFLAGS="-s"
 
 AM_DISABLE_SHARED
